int/api/cmd/deploySC.go

```diff
@@ -22,45 +26,69 @@ type deploySC struct {
 }
 
 func (d *deploySC) Handle(params operations.CmdDeploySCParams) middleware.Responder {
-	file, err := io.ReadAll(params.SmartContract)
+	smartContractByteCode, err := base64.StdEncoding.DecodeString(params.Body.SmartContract)
 	if err != nil {
-		return operations.NewCmdDeploySCBadRequest().
+		return operations.NewCmdDeploySCInternalServerError().
 			WithPayload(
 				&models.Error{
 					Code:    err.Error(),
 					Message: err.Error(),
 				})
 	}
-	/* All the pointers below cannot be null as the swagger hydrate
-	each one with their default value defined in swagger.yml,
-	if no values are provided for these parameters.
-	*/
-	decodedDatastore, err := base64.StdEncoding.DecodeString(*params.Datastore)
+
+	parameters, err := base64.StdEncoding.DecodeString(params.Body.Parameters)
 	if err != nil {
-		return operations.NewCmdDeploySCBadRequest().
+		return operations.NewCmdDeploySCInternalServerError().
```

Comment:
@thomas-senechal  which is best NewCmdDeploySCBadRequest() or NewCmdDeploySCInternalServerError() ?

Reply:
`NewCmdDeploySCBadRequest`
It's VERY LIKELY an issue with the request if we can't decode the base64